tests/docker_cluster.py

```diff
@@ -334,67 +336,80 @@ def exec_cmd_on_host(self, host, cmd, raise_error=True, tty=False):
+        slave_name = dc._get_slave_image_name(dc.BARE_CLUSTER_TYPE)
+        centos_cluster = DockerCluster('master',
+                                       ['slave1', 'slave2', 'slave3'],
+                                       DEFAULT_LOCAL_MOUNT_POINT,
+                                       DEFAULT_DOCKER_MOUNT_POINT)
+
+        if not dc._check_for_images(master_name, slave_name):
+            centos_cluster.create_image(
+                BASE_TD_DOCKERFILE_DIR,
+                master_name,
+                BASE_IMAGE_NAME,
+                BASE_IMAGE_TAG
             )
-            presto_cluster.client.commit(
-                presto_cluster.slaves[0],
-                INSTALLED_PRESTO_TEST_SLAVE_IMAGE
+
+            centos_cluster.create_image(
+                BASE_TD_DOCKERFILE_DIR,
+                slave_name,
+                BASE_IMAGE_NAME,
+                BASE_IMAGE_TAG
             )
-            return presto_cluster
+
+        centos_cluster.start_containers(master_name, slave_name)
+
+        return centos_cluster
 
     @staticmethod
-    def start_base_cluster():
+    def start_existing_images(cluster_type):
+        DC = DockerCluster
+        master_name = DC._get_master_image_name(cluster_type)
+        slave_name = DC._get_slave_image_name(cluster_type)
+
+        if not DC._check_for_images(master_name, slave_name):
+            return None
+
         centos_cluster = DockerCluster('master',
                                        ['slave1', 'slave2', 'slave3'],
                                        DEFAULT_LOCAL_MOUNT_POINT,
                                        DEFAULT_DOCKER_MOUNT_POINT)
-        centos_cluster.create_image(
-            os.path.join(LOCAL_RESOURCES_DIR, 'centos6-ssh-test'),
-            'teradatalabs/centos6-ssh-test',
-            'jdeathe/centos-ssh',
-            'centos-6-1.2.0'
-        )
-        centos_cluster.start_containers(
-            'teradatalabs/centos6-ssh-test',
-            'teradatalabs/centos6-ssh-test'
-        )
+
+        centos_cluster.start_containers(master_name, slave_name)
         return centos_cluster
 
     @staticmethod
-    def _check_for_presto_images():
+    def _check_for_images(master_image_name, slave_image_name):
         client = Client(timeout=180)
         images = client.images()
         has_master_image = False
         has_slave_image = False
         for image in images:
-            if INSTALLED_PRESTO_TEST_MASTER_IMAGE in image['RepoTags'][0]:
+            if master_image_name in image['RepoTags'][0]:
                 has_master_image = True
-            if INSTALLED_PRESTO_TEST_SLAVE_IMAGE in image['RepoTags'][0]:
+            if slave_image_name in image['RepoTags'][0]:
                 has_slave_image = True
         return has_master_image and has_slave_image
 
+    def commit_images(self, cluster_type):
+        self.client.commit(self.master,
+                           self._get_master_image_name(cluster_type))
```

Comment:
make sure to run lint, this is a bit suspicious indentation (though it could just be github)